File: Api/services/interfaces/core/IStylePropertyService.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel # Assuming services might use Pydantic models
from abc import ABC, abstractmethod # For interfaces

class IStylePropertyService(ABC):
    @abstractmethod
    def add_style_property(self, style_property: object # TODO: Specify correct type hint):
        pass

    @abstractmethod
    def update_style_property(self, style_property: object # TODO: Specify correct type hint):
        pass

    @abstractmethod
    def delete_style_property(self, id: object # TODO: Specify correct type hint):
        pass

File: Framework.Services/Interfaces/Core/IStylePropertyService.cs
using Framework.Data.Models.Conteudo;
using System.Collections.Generic;

namespace Framework.Services.Interfaces.Core
{
    public interface IStylePropertyService
    {
        IEnumerable<StyleProperty> GetStyleProperties(int ownerId, string ownerType);
        void AddStyleProperty(StyleProperty styleProperty);
        void UpdateStyleProperty(StyleProperty styleProperty);
        void DeleteStyleProperty(int id);
    }

}
